provider/datagen/src/cldr/transform/time_zones/convert.rs

```diff
@@ -185,21 +189,31 @@ macro_rules! long_short_impls {
                                     key.push('/');
                                     key.push_str(&inner_key);
                                     match place_or_region {
-                                        LocationOrSubRegion::Location(place) => place
-                                            .$metazones_name()
-                                            .and_then(|zf| type_fallback(&zf).cloned())
-                                            .map(|format| vec![(key, format)])
-                                            .unwrap_or_default(),
+                                        LocationOrSubRegion::Location(place) => {
+                                            match bcp47_tzid_data.get(&key) {
+                                                Some(bcp47) => place
+                                                    .$metazones_name()
+                                                    .and_then(|zf| type_fallback(&zf).cloned())
+                                                    .map(|format| vec![(bcp47, format)])
+                                                    .unwrap_or_default(),
+                                                None => panic!("Cannot find bcp47 for {:?}.", key),
+                                            }
+                                        }
                                         LocationOrSubRegion::SubRegion(region) => region
                                             .iter()
                                             .filter_map(|(inner_key, place)| {
                                                 let mut key = key.clone();
                                                 key.push('/');
                                                 key.push_str(&inner_key);
-                                                place
-                                                    .$metazones_name()
-                                                    .and_then(|zf| type_fallback(&zf).cloned())
-                                                    .map(|format| (key, format))
+                                                match bcp47_tzid_data.get(&key) {
```

Comment:
I forgot to update here in my previous PRs. And I believe this change updated json files in this PR.